### ISSUE TITLE
CP 2.2.0 seg faults on Ubuntu 16.04.2 LTS

### ISSUE THREAD

**Issue by dmattek** [CellProfiler/CellProfiler#2798]
Everything was working fine until latest apt-get upgrade. 

Right after:

`Caused by: java.io.IOException: invalid constant type: 18`

I'm getting:

`Segmentation fault (core dumped)`

The output from /var/log/apport.log:

`ERROR: apport (pid 24998) Wed Jun 21 17:29:31 2017: called for pid 24971, signal 11, core limit 0`
`ERROR: apport (pid 24998) Wed Jun 21 17:29:31 2017: script: /home/xxx/.local/bin/cellprofiler, interpreted by /usr/bin/python2.7 (command line "/usr/bin/python /home/xxx/.local/bin/cellprofiler")`
`ERROR: apport (pid 24998) Wed Jun 21 17:29:31 2017: executable does not belong to a package, ignoring`

Update after which the seg fault started:

> Commandline: apt-get upgrade
Requested-By: xxxxxxxxxx
Upgrade: postgresql-client-9.5:amd64 (9.5.6-0ubuntu0.16.04, 9.5.7-0ubuntu0.16.04), libqt5test5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), whoopsie:amd64 (0.2.52.2, 0.2.52.3), openssl:amd64 (1.0.2g-1ubuntu4.6, 1.0.2g-1ubuntu4.8), vino:amd64 (3.8.1-0ubuntu9.1, 3.8.1-0ubuntu9.2), grub-common:amd64 (2.02~beta2-36ubuntu3.9, 2.02~beta2-36ubuntu3.11), snapd:amd64 (2.24.1, 2.25), libwhoopsie0:amd64 (0.2.52.2, 0.2.52.3), libnm-gtk-common:amd64 (1.2.6-0ubuntu0.16.04.2, 1.2.6-0ubuntu0.16.04.3), libqt5dbus5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), libqt5sql5-sqlite:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), libqt5widgets5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), grub2-common:amd64 (2.02~beta2-36ubuntu3.9, 2.02~beta2-36ubuntu3.11), libepoxy0:amd64 (1.3.1-1, 1.3.1-1ubuntu0.16.04.1), network-manager-gnome:amd64 (1.2.6-0ubuntu0.16.04.2, 1.2.6-0ubuntu0.16.04.3), isc-dhcp-common:amd64 (4.3.3-5ubuntu12.6, 4.3.3-5ubuntu12.7), libpq5:amd64 (9.5.6-0ubuntu0.16.04, 9.5.7-0ubuntu0.16.04), grub-pc:amd64 (2.02~beta2-36ubuntu3.9, 2.02~beta2-36ubuntu3.11), libqt5xml5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), libqt5printsupport5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), grub-pc-bin:amd64 (2.02~beta2-36ubuntu3.9, 2.02~beta2-36ubuntu3.11), apport:amd64 (2.20.1-0ubuntu2.5, 2.20.1-0ubuntu2.6), libqt5gui5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), postgresql-9.5:amd64 (9.5.6-0ubuntu0.16.04, 9.5.7-0ubuntu0.16.04), libmirprotobuf3:amd64 (0.21.0+16.04.20160330-0ubuntu1, 0.26.3+16.04.20170605-0ubuntu1), python3-apport:amd64 (2.20.1-0ubuntu2.5, 2.20.1-0ubuntu2.6), libnma0:amd64 (1.2.6-0ubuntu0.16.04.2, 1.2.6-0ubuntu0.16.04.3), libnma-common:amd64 (1.2.6-0ubuntu0.16.04.2, 1.2.6-0ubuntu0.16.04.3), libqt5core5a:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), postgresql-contrib-9.5:amd64 (9.5.6-0ubuntu0.16.04, 9.5.7-0ubuntu0.16.04), iproute2:amd64 (4.3.0-1ubuntu3, 4.3.0-1ubuntu3.16.04.1), libnm-gtk0:amd64 (1.2.6-0ubuntu0.16.04.2, 1.2.6-0ubuntu0.16.04.3), libqt5opengl5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), apport-gtk:amd64 (2.20.1-0ubuntu2.5, 2.20.1-0ubuntu2.6), unattended-upgrades:amd64 (0.90ubuntu0.5, 0.90ubuntu0.6), linux-firmware:amd64 (1.157.8, 1.157.11), libqt5network5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), isc-dhcp-client:amd64 (4.3.3-5ubuntu12.6, 4.3.3-5ubuntu12.7), python3-problem-report:amd64 (2.20.1-0ubuntu2.5, 2.20.1-0ubuntu2.6), libqt5sql5:amd64 (5.5.1+dfsg-16ubuntu7.2, 5.5.1+dfsg-16ubuntu7.5), libssl1.0.0:amd64 (1.0.2g-1ubuntu4.6, 1.0.2g-1ubuntu4.8)

**Comment by 0x00b1** [comment]
Would you mind posting the output of `pip freeze`?

**Comment by dmattek** [comment]
`pip freeze`

> adium-theme-ubuntu==0.3.4
cellh5==1.3.0
-e git+https://github.com/CellProfiler/CellProfiler.git@b5b4485d5ac12476ba088003cf0081eb4e3b0d82#egg=CellProfiler
centrosome==1.0.5
contextlib2==0.5.5
cycler==0.9.0
Cython==0.23.4
dask==0.12.0
decorator==4.0.6
dnspython==1.12.0
h5py==2.6.0
hmmlearn==0.2.0
inflect==0.2.5
javabridge==1.0.14
joblib==0.11
libtiff==0.4.0
lxml==3.6.4
mahotas==1.4.3
matplotlib==1.5.1
MySQL-python==1.2.5
mysqlclient==1.3.7
networkx==1.11
numpy==1.11.0
pandas==0.19.1
Pillow==3.1.2
prokaryote==1.0.11
py==1.4.31
pyamg==3.1.1
pycrypto==2.6.1
PyMySQL==0.7.9
pyparsing==2.0.3
pytest==3.0.3
python-bioformats==1.0.9
python-dateutil==2.4.2
pytz==2014.10
pyzmq==15.2.0
raven==6.0.0
requests==2.14.2
scikit-image==0.12.3
scikit-learn==0.18
scipy==0.17.0
scour==0.32
six==1.10.0
toolz==0.8.0
unity-lens-photos==1.0
vboxapi==1.0
wxPython==3.0.2.0
wxPython-common==3.0.2.0

**Comment by 0x00b1** [comment]
Can you try using master? 

```sh
pip uninstall cellprofiler
pip install git+https://github.com/CellProfiler/CellProfiler.git
```

**Comment by dmattek** [comment]
Also segmentation fault with 3.0.0rc1

**Comment by 0x00b1** [comment]
Yikes. Would you mind posting all of the CellProfiler logs (in addition to the `Caused by: java.io.IOException: invalid constant type: 18` exception)?

**Comment by dmattek** [comment]
`cellprofiler --version`
`CellProfiler 2.2.0`
`Git ac0529e`
`Version 20160503183100`
`Built 2016-05-03T18:31:00`

`cellprofiler`
`Plugin directory doesn't point to valid folder: /home/xxx/plugins`
`java.lang.IllegalArgumentException: Cannot handle replace call to list in ij.Menus's public static synchronized java.lang.String[] getPlugins()
	at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:725)
	at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:650)
	at net.imagej.patcher.LegacyExtensions.addExtraPlugins(LegacyExtensions.java:477)
	at net.imagej.patcher.LegacyExtensions.injectHooks(LegacyExtensions.java:175)
	at net.imagej.patcher.LegacyInjector.inject(LegacyInjector.java:308)
	at net.imagej.patcher.LegacyInjector.injectHooks(LegacyInjector.java:109)
	at net.imagej.patcher.LegacyEnvironment.initialize(LegacyEnvironment.java:101)
	at net.imagej.patcher.LegacyEnvironment.applyPatches(LegacyEnvironment.java:495)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:397)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:376)
	at net.imagej.patcher.JavaAgent.preinit(JavaAgent.java:147)
	at net.imagej.patcher.JavaAgent.premain(JavaAgent.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
	at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.RuntimeException: java.io.IOException: invalid constant type: 18
	at javassist.CtClassType.getClassFile2(CtClassType.java:203)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:110)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:96)
	at javassist.compiler.TypeChecker.atMethodCallCore(TypeChecker.java:704)
	at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:681)
	at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:156)
	at javassist.compiler.ast.CallExpr.accept(CallExpr.java:45)
	at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:241)
	at javassist.compiler.CodeGen.atStmnt(CodeGen.java:329)
	at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
	at javassist.compiler.CodeGen.atIfStmnt(CodeGen.java:390)
	at javassist.compiler.CodeGen.atStmnt(CodeGen.java:354)
	at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
	at javassist.compiler.Javac.compileStmnt(Javac.java:568)
	at javassist.expr.MethodCall.replace(MethodCall.java:234)
	at net.imagej.patcher.CodeHacker$7.edit(CodeHacker.java:686)
	at javassist.expr.ExprEditor.loopBody(ExprEditor.java:191)
	at javassist.expr.ExprEditor.doit(ExprEditor.java:90)
	at javassist.CtBehavior.instrument(CtBehavior.java:618)
	at net.imagej.patcher.CodeHacker$EagerExprEditor.instrument(CodeHacker.java:1278)
	at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:670)
	... 17 more
Caused by: java.io.IOException: invalid constant type: 18
	at javassist.bytecode.ConstPool.readOne(ConstPool.java:1027)
	at javassist.bytecode.ConstPool.read(ConstPool.java:970)
	at javassist.bytecode.ConstPool.<init>(ConstPool.java:127)
	at javassist.bytecode.ClassFile.read(ClassFile.java:693)
	at javassist.bytecode.ClassFile.<init>(ClassFile.java:85)
	at javassist.CtClassType.getClassFile2(CtClassType.java:190)
	... 37 more
java.lang.IllegalArgumentException: Cannot handle app name in ij.ImageJ's public <init>(java.applet.Applet applet, int mode)
	at net.imagej.patcher.CodeHacker.replaceAppNameInCall(CodeHacker.java:446)
	at net.imagej.patcher.LegacyExtensions.insertAppNameHooks(LegacyExtensions.java:406)
	at net.imagej.patcher.LegacyExtensions.injectHooks(LegacyExtensions.java:291)
	at net.imagej.patcher.LegacyInjector.inject(LegacyInjector.java:308)
	at net.imagej.patcher.LegacyInjector.injectHooks(LegacyInjector.java:109)
	at net.imagej.patcher.LegacyEnvironment.initialize(LegacyEnvironment.java:101)
	at net.imagej.patcher.LegacyEnvironment.applyPatches(LegacyEnvironment.java:495)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:397)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:376)
	at net.imagej.patcher.JavaAgent.preinit(JavaAgent.java:147)
	at net.imagej.patcher.JavaAgent.premain(JavaAgent.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
	at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.RuntimeException: java.io.IOException: invalid constant type: 18
	at javassist.CtClassType.getClassFile2(CtClassType.java:203)
	at javassist.CtClassType.subtypeOf(CtClassType.java:303)
	at javassist.CtClassType.subtypeOf(CtClassType.java:318)
	at javassist.compiler.MemberResolver.compareSignature(MemberResolver.java:247)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:119)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:96)
	at javassist.compiler.TypeChecker.atMethodCallCore(TypeChecker.java:704)
	at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:681)
	at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:156)
	at javassist.compiler.ast.CallExpr.accept(CallExpr.java:45)
	at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:241)
	at javassist.compiler.CodeGen.atStmnt(CodeGen.java:329)
	at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
	at javassist.compiler.Javac.compileStmnt(Javac.java:568)
	at javassist.expr.MethodCall.replace(MethodCall.java:234)
	at net.imagej.patcher.CodeHacker$4.edit(CodeHacker.java:427)
	at net.imagej.patcher.CodeHacker$4.edit(CodeHacker.java:441)
	at javassist.expr.ExprEditor.loopBody(ExprEditor.java:219)
	at javassist.expr.ExprEditor.doit(ExprEditor.java:90)
	at javassist.CtBehavior.instrument(CtBehavior.java:618)
	at net.imagej.patcher.CodeHacker$EagerExprEditor.instrument(CodeHacker.java:1278)
	at net.imagej.patcher.CodeHacker.replaceAppNameInCall(CodeHacker.java:402)
	... 16 more
Caused by: java.io.IOException: invalid constant type: 18
	at javassist.bytecode.ConstPool.readOne(ConstPool.java:1027)
	at javassist.bytecode.ConstPool.read(ConstPool.java:970)
	at javassist.bytecode.ConstPool.<init>(ConstPool.java:127)
	at javassist.bytecode.ClassFile.read(ClassFile.java:693)
	at javassist.bytecode.ClassFile.<init>(ClassFile.java:85)
	at javassist.CtClassType.getClassFile2(CtClassType.java:190)
	... 37 more`
`Segmentation fault (core dumped)`

**Comment by samfux84** [comment]
This is a problem caused by the Prokaryote package on which CellProfiler depends on. The version of ImageJ bundled with Prokaryote is too old.

The problem is discussed in a issue in the ImageJ github repository:

https://github.com/imagej/tutorials/issues/24

Their proposed solution is to upgrade ImageJ.

Back in September 2016 I have opened an issue in the CellProfiler/Prokaryote github repository, asking if it would be possible to bundle a newer version of ImageJ with Prokaryote:

https://github.com/CellProfiler/prokaryote/issues/24

Up to know I did not get an answer to this issue.

**Comment by dmattek** [comment]
I was getting these warnings before but still CP started and worked fine. It's only now after `apt-get upgrade` I'm getting the seg fault.

**Comment by samfux84** [comment]
You are right. Back in September 2016, when I reported the problem, there was no segfault, just warnings. But if there is a new problem, it would be easier to investigate if the logs were not flooded with Java warnings, especially since it is known how to get rid of them.

On our HPC cluster, I have installed CellProfiler 2.2 and it used to work fine. But today some users reported that their CellProfiler jobs crash with a SIGBUS error. We did not update CellProfiler and there were only small changes in the operating system (moving from CentOS 6.3 to 6.8). But I can not exclude that the problem is related to a change in a system library.

**Comment by dmattek** [comment]
@samfux84 Would you mind checking updated packages in your system and whether there's an overlap with the list I posted earlier?

**Comment by 0x00b1** [comment]
@samfux84 thanks for the feedback and solution. We reworked Prokaryote to resolve this (and related) problems. The solution we found that worked was unbundling ImageJ. It should work with whatever recent version of ImageJ that’s available (and on your path). I would love to hear your feedback.

**Comment by samfux84** [comment]
@0x00b1 Thank you very much for reworking Prokaryote. This is appreciated. Unbundling ImageJ is a good solution as it allows to change the ImageJ version without changing the Prokaryote version.

@dmattek I got some more feedback from our users that received the SIGBUS error. On Friday 23. June we updated some packages due to the security vulnerabilities in the Linux kernel (CVE-2017-1000364) and glibc (CVE-2017-1000366). We therefore updated the kernel and glibc packages. At the moment I don't have the full list of packages and our system administrator is out-of-office today. I will ask her tomorrow.

Before this update, CellProfiler 2.2.0 was working without SIGBUS error and after the security upgrade it was crashing with errors (according to reports from our users). I tested it on a patched compute node vs. an unpatched one.

Currently I am working on providing the list of changed packages.

**Comment by samfux84** [comment]
Please find below a list of all packages (CentOS 6.9) that were updated in the security fix (the list shows for all packages first the old version of the unpatched system and afterwards the new version on the patched system):

```
< ca-certificates.noarch 2016.2.10-65.4.el6
> ca-certificates.noarch 2017.2.14-65.0.1.el6_9
< cups-libs.x86_64       1:1.4.2-77.el6   
> cups-libs.x86_64       1:1.4.2-78.el6_9               
< glibc.x86_64           2.12-1.209.el6_9.1
> glibc.x86_64           2.12-1.209.el6_9.2
< glibc-common.x86_64    2.12-1.209.el6_9.1
> glibc-common.x86_64    2.12-1.209.el6_9.2
< glibc-devel.x86_64     2.12-1.209.el6_9.1
> glibc-devel.x86_64     2.12-1.209.el6_9.2
< glibc-headers.x86_64   2.12-1.209.el6_9.1
> glibc-headers.x86_64   2.12-1.209.el6_9.2
< initscripts.x86_64     9.03.58-1.el6.centos
> initscripts.x86_64     9.03.58-1.el6.centos.1                
< iser.x86_64            4.0-OFED.4.0.2.0.0.1.g595cb61.kver.2.6.32_696.1.1.el6.euler.x86_64
> iser.x86_64            4.0-OFED.4.0.2.0.0.1.g595cb61.kver.2.6.32_696.3.2.el6.euler.x86_64
< kernel.x86_64          2.6.32-696.1.1.el6.euler
> kernel.x86_64          2.6.32-696.3.2.el6.euler
< kernel-devel.x86_64    2.6.32-696.1.1.el6.euler
> kernel-devel.x86_64    2.6.32-696.3.2.el6.euler
< kernel-firmware.noarch 2.6.32-696.1.1.el6.euler
> kernel-firmware.noarch 2.6.32-696.3.2.el6.euler
< kernel-headers.x86_64  2.6.32-696.1.1.el6.euler
> kernel-headers.x86_64  2.6.32-696.3.2.el6.euler
< kernel-mft.x86_64      4.6.0-51.kver.2.6.32_696.1.1.el6.euler.x86_64
> kernel-mft.x86_64      4.6.0-51.kver.2.6.32_696.3.2.el6.euler.x86_64
< knem.x86_64            1.1.2.90mlnx2-OFED.4.0.1.6.3.1.g4faa297.kver.2.6.32_696.1.1.el6.euler.x86_64
> knem.x86_64            1.1.2.90mlnx2-OFED.4.0.1.6.3.1.g4faa297.kver.2.6.32_696.3.2.el6.euler.x86_64
< ksh.x86_64             20120801-33.el6  
> ksh.x86_64             20120801-34.el6_9
< 4.0-OFED.4.0.2.0.0.1.g595cb61.kver.2.6.32_696.1.1.el6.euler.x86_64
> 4.0-OFED.4.0.2.0.0.1.g595cb61.kver.2.6.32_696.3.2.el6.euler.x86_64
< nss.x86_64             3.28.4-1.el6_9           
> nss.x86_64             3.28.4-3.el6_9            
< nss-sysinit.x86_64     3.28.4-1.el6_9          
> nss-sysinit.x86_64     3.28.4-3.el6_9
< nss-tools.x86_64       3.28.4-1.el6_9                                    
> nss-tools.x86_64       3.28.4-3.el6_9                  
< rpcbind.x86_64         0.2.0-13.el6_9              
> rpcbind.x86_64         0.2.0-13.el6_9.1        
< sudo.x86_64            1.8.6p3-27.el6
> sudo.x86_64            1.8.6p3-29.el6_9  
```

**Comment by dmattek** [comment]
@samfux84 Thanks for your input. In my case the culprit seems to be the following update:

`Start-Date: 2017-06-20  05:07:01`
`Commandline: /usr/bin/unattended-upgrade`
`Install: linux-image-4.4.0-81-generic:amd64 (4.4.0-81.104, automatic), linux-headers-4.4.0-81:amd64 (4.4.0-81.104, automatic), linux-image-extra-4.4.0-81-generic:amd64 (4.4.0-81.1
04, automatic), linux-headers-4.4.0-81-generic:amd64 (4.4.0-81.104, automatic)`
`Upgrade: linux-headers-generic:amd64 (4.4.0.79.85, 4.4.0.81.87), linux-libc-dev:amd64 (4.4.0-79.100, 4.4.0-81.104), libc6-dbg:amd64 (2.23-0ubuntu7, 2.23-0ubuntu9), libc6-dev:amd64
 (2.23-0ubuntu7, 2.23-0ubuntu9), linux-image-generic:amd64 (4.4.0.79.85, 4.4.0.81.87), libc6:amd64 (2.23-0ubuntu7, 2.23-0ubuntu9), locales:amd64 (2.23-0ubuntu7, 2.23-0ubuntu9), li
bc-bin:amd64 (2.23-0ubuntu7, 2.23-0ubuntu9), libc-dev-bin:amd64 (2.23-0ubuntu7, 2.23-0ubuntu9), multiarch-support:amd64 (2.23-0ubuntu7, 2.23-0ubuntu9), linux-generic:amd64 (4.4.0.
79.85, 4.4.0.81.87)`
`End-Date: 2017-06-20  05:08:28`

Once rebooted with:
`gnulinux-4.4.0-79-generic-advanced`

CP2.2.0 and CP3.0.0rc1 work fine as before (+ the usual stream of Java warnings).

**Comment by samfux84** [comment]
@dmattek Thank you for your feedback. In my case, we cannot go back, as the updates were done to fix a security vulnerability. In my opinion software should also work fine with the slightly newer kernel (it was no major upgrade).

**Comment by anou85** [comment]
Hi, 
sorry for joining in, but  I'm actually having the same problem. CP was working fine until a few days ago, after I did an apt-get upgrade.
When I try to use the pip unistall as suggested before, I'm getting: 
```
system > pip uninstall cellprofiler
Cannot uninstall requirement cellprofiler, not installed
```
the `pip install --upgrade git+https://github.com/CellProfiler/CellProfiler.git`
gives me `Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-OuSmb7-build/`
I wouldn't even know which libraries were updated during the upgrade as the `/usr/bin/unattended-upgrade` shows nothing...

**Comment by dmattek** [comment]
@anou85 I wouldn't uninstall CP. A temporary solution that works for me is to reboot using an earlier kernel (`gnulinux-4.4.0-79-generic-advanced`). You should be able to select it at boot time in GRUB (`Advanced options for Ubuntu`) or make this the default kernel by editing GRUB_DEFAULT option in `/etc/default/grub` ([https://unix.stackexchange.com/a/327686/185359](url)).

You can check your updated packages by viewing:
`less /var/log/apt/history.log`

**Comment by anou85** [comment]
perfect, it works booting from an earlier kernel! thank you very much.

**Comment by samfux84** [comment]
@dmattek @0x00b1 @anou85 Please not that running an earlier kernel solves the problem of CellProfiler crashing, but be aware that you run a buggy kernel which makes your system insecure.

At the moment the proper way to deal with this issue would be to update the kernel to the safe version 4.4.0-83 and use a workaround to avoid the crash.

According to https://askubuntu.com/questions/927746/eclipse-crashes-with-linux-kernel-4-4-0-81-generic the problem can be avoided, when increasing the JVM max stack size using -Xss1280k when starting Java. I tested it on our CentOS 6.9 system and the workaround avoids the crash.

The easiest way to do this is to use the following command before starting CellProfiler:

export JAVA_TOOL_OPTIONS=-Xss1280k

I will add this command directly to the CellProfiler environment module that our users load before running the program.

**Comment by imagejan** [comment]
On the ImageJ forum, a similar issue related to Ubuntu 16.04 and segmentation fault was reported and is being discussed [here](http://forum.imagej.net/t/unable-to-start-imagej-on-ubuntu-16-04-2-fresh-install/5757?u=imagejan).

**Comment by AetherUnbound** [comment]
Is this still an issue with the latest kernel version?

**Comment by mcquin** [comment]
Hi @dmattek & co,

We solved this issue today on an AWS instance of Ubuntu 16.04 by installing the Oracle JDK, following these instructions: https://www.digitalocean.com/community/tutorials/how-to-install-java-with-apt-get-on-ubuntu-16-04

```
sudo add-apt-repository ppa:webupd8team/java
sudo apt-get update
sudo apt-get install oracle-java8-installer
```

Hope you also find this helpful!

**Comment by anou85** [comment]
Haven't had any issues with the latest kernel version but it's running on Oracle JDK anyways

**Comment by mcquin** [comment]
Thanks for the update @anou85. We'll close this until another issue is reported.
